### PR TITLE
GEOPY-1440: geoh5py accepot any kind of integer

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-1440 - geoh5py accepot any kind of integer**
change int32 to np.integer to prevent bugs if np.int64 is used

### CODE REVIEW

File: geoh5py/data/data_type.py [hunk 1]
```diff
@@ -319,13 +319,14 @@ def validate_data_type(workspace: Workspace, attribute_dict: dict):
             else:
                 values = attribute_dict.get("values")
                 if values is None or (
-                    isinstance(values, np.ndarray)
-                    and (values.dtype in [np.float32, np.float64])
+                    isinstance(values, np.ndarray) and (values.dtype in [np.floating])
```
Comment (domfournier) [inline geoh5py/data/data_type.py:322]:
```suggestion
                    isinstance(values, np.ndarray) and np.issubdtype(values.dtype, np.floating)
```

File: geoh5py/groups/group.py [hunk 2]
```diff
@@ -45,6 +46,22 @@ def __init__(self, group_type: GroupType, **kwargs):
     def default_type_uid(cls) -> uuid.UUID | None:
         """Abstract method to return the default type uid for the class."""
 
+    def add_children(self, children: list[Entity]):
```
[the remark this answers is not in the document]
Comment (MatthieuCMira) [reply geoh5py/groups/group.py]: not concerning this PR

File: geoh5py/ui_json/templates.py [hunk 3]
```diff
@@ -448,3 +449,58 @@ def drillhole_group_data(
     if optional is not None:
         form.update(optional_parameter(optional))
     return form
+
+
+def range_label_template(
```
[the remark this answers is not in the document]
Comment (MatthieuCMira) [reply geoh5py/ui_json/templates.py]: not concerning this PR

File: geoh5py/shared/concatenation/object.py [hunk 4]
```diff
@@ -98,14 +73,11 @@ def create_property_group(
 
         return prop_group
 
-    def get_data(self, name: str | uuid.UUID) -> list[Data]:
+    def _fetch_concatenated_children(self):
```
[the remark this answers is not in the document]
Comment (MatthieuCMira) [reply geoh5py/shared/concatenation/object.py]: not concerning this PR

**Review by domfournier — APPROVED** [review]
Should be good. Thanks

**Pull request merged**